monoio/src/runtime.rs

```diff
@@ -382,6 +379,40 @@ where
     join
 }
 
+/// Similar to [`spawn()`], but it catches any panic that occurs in the spawned task.
+///
+/// Note that `.await`ing returned `JoinHandle` now returns a `Result<T::Output>`
+/// rather than `T::Output`.
+///
+/// # Examples
+///
+/// ```no_run
+/// #[monoio::main]
+/// async fn main() {
+///     let handle = monoio::spawn_catch_unwind(async {
+///         println!("hello from a background task");
+///     });
+///
+///     // Let the task complete
+///     handle.await.expect("no panic would happen!");
+/// }
+/// ```
+pub fn spawn_catch_unwind<T>(
+    future: T,
+) -> JoinHandle<Result<T::Output, Box<dyn std::any::Any + 'static>>>
+where
+    T: Future + 'static,
+    T::Output: 'static,
+{
+    use futures::{FutureExt, TryFutureExt};
```

Comment:
Using `futures_util` instead of `futures` can reduce dependencies. Or fork `CatchUnwind` can avoid introducing new dependency.

Reply:
>  Or fork CatchUnwind can avoid introducing new dependency.

Yeah, this makes more sense

Reply:
Could you change in any of these 2 ways?
Note if you choose to fork code, remember to add enough note for the copyright.

Reply:
> Could you change in any of these 2 ways?

Yeah, I will, I am just busy with my work.

> Note if you choose to fork code, remember to add enough note for the copyright.

Get it.